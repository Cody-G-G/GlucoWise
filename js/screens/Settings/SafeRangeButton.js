'use strict';
import React, {Component} from 'react';
import {TouchableOpacity, Text, StyleSheet} from 'react-native';
import styles from "./styles";

export default class SafeRangeButton extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <TouchableOpacity
                style={StyleSheet.flatten([styles.safeRangeButton, {backgroundColor: this.props.backgroundColor}])}
                onPress={() => this.props.handlePress()}>
                <Text style={styles.safeRangeButtonText}>{this.props.buttonText}</Text>
            </TouchableOpacity>
        );
    }
}