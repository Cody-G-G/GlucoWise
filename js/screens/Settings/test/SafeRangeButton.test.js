'use strict';
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import SafeRangeButton from "../SafeRangeButton";

test('SafeRangeButton - renders correctly', () => {
    const backgroundColor = 'royalblue';
    const buttonText = 'Set';

    const tree = renderer.create(
        <SafeRangeButton backgroundColor={backgroundColor}
                         buttonText={buttonText}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});