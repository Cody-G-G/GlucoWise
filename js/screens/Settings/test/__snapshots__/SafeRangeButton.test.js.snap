// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeRangeButton - renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "royalblue",
      "borderWidth": 0.5,
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 23,
        "fontWeight": "bold",
      }
    }
  >
    Set
  </Text>
</View>
`;
